source/opt/eliminate_dead_input_components_pass.cpp

```diff
@@ -179,7 +210,11 @@ void EliminateDeadInputComponentsPass::ChangeStructLength(
   analysis::TypeManager* type_mgr = context()->get_type_mgr();
   analysis::Pointer* ptr_type =
       type_mgr->GetType(struct_var.type_id())->AsPointer();
-  const analysis::Struct* struct_ty = ptr_type->pointee_type()->AsStruct();
+  auto core_type = ptr_type->pointee_type();
+  // Check for array of struct from tesc, tese and geom
+  const auto arr_type = core_type->AsArray();
+  if (arr_type) core_type = arr_type->element_type();
+  const analysis::Struct* struct_ty = core_type->AsStruct();
```

Comment:
missed this in the first review, sorry.
- this is called ChangeStructType, and was a straightforward function: take a struct, change its length.
- now it's doing some magic with arrays, allowing one level of array top be skipped. I understand the goal in this context, but don't think this should be done here.

What about:
 - this function takes a struct type, and returns a new type with the proper size.
 - outside, you do struct_var->setResultType(resized_type);
 
 This way, the function remains true to its name, and doesn't have some less known behaviors.

Reply:
Good point. But rather than change the function to be true to the name, I decided to change the name :)

I changed the name to ChangeIOVarStructLength, which is to say, change the (outermost) struct length in the IO var. I also renamed the argument from struct_var to io_var to reflect that the var may not be be a struct (eg a per-vertex-array of struct). I also improved the header comment of this function to explain this all.